### PR TITLE
[BUG] Force `softmax` returning Series

### PR DESCRIPTION
**Pull request by Zeroto521**
# PR Description

Please describe the changes proposed in the pull request:

```python
115         >>> import pandas as pd
116         >>> import janitor
117         >>> s = pd.Series([0, 1, 3], name="numbers")
118         >>> s.softmax()
Expected:
    0    0.042010
    1    0.114195
    2    0.843795
    Name: numbers, dtype: float64
Got:
    array([0.04201007, 0.1141952 , 0.84379473])
```

**This PR resolves https://github.com/pyjanitor-devs/pyjanitor/runs/7703616556?check_suite_focus=true .**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.md`.

3. [x] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

File: janitor/math.py [hunk 1]
```diff
@@ -124,7 +124,8 @@ def softmax(s: pd.Series) -> pd.Series:
     :param s: Input Series.
     :return: Transformed Series.
     """
-    return scipy_softmax(s)
+
+    return pd.Series(scipy_softmax(s), index=s.index, name=s.name)
```
Comment (Zeroto521) [inline janitor/math.py:128]: Actually I'd like use `np.exp(s) / sum(np.exp(s))` instead of using scipy

**Review by samukweku — APPROVED** [review]

File: janitor/math.py (hunk 1 above)
Comment (ericmjl) [reply janitor/math.py:128]: I think the SciPy implementation comes with numerical stability checks for free. Is there a reason why you would prefer to use a NumPy implementation?

**Review by ericmjl — APPROVED** [review]
I am good to go. I think we should keep the SciPy soft Mac implementation rather than switch to numpy, for numerical stability reasons (IIRC).

File: janitor/math.py (hunk 1 above)
Comment (Zeroto521) [reply janitor/math.py:128]: To totally replace scipy. Since we only use it in `math.py` and `functions/impute.py`
- `mode` function in `functions/impute.py` could replace by `Series.value_counts`
- `math.py`
  - `sigmoid` (`expit`) could use `np.exp`
  - `softmax` (`scipy.special.softmax`) could use `np.exp`
  - `logit` (`scipy.special.softmax`) could use `np.log`
  - `normal_cdf` (`scipy.stats.norm.cdf`) and `probit` (`scipy.stats.norm.ppf`) can't replace easily.
